oak_restricted_kernel/src/mm/mod.rs

```diff
@@ -150,9 +167,19 @@ pub fn init_paging<A: FrameAllocator<Size4KiB> + ?Sized>(
     let pml4 = unsafe { &mut *(pml4_frame.start_address().as_u64() as *mut PageTable) };
     pml4.zero();
 
+    // Should we set the C-bit (encrypted memory for SEV)? For now, let's assume it's bit 51.
+    let encrypted = if get_sev_status()
+        .unwrap_or(SevStatus::empty())
+        .contains(SevStatus::SEV_ENABLED)
+    {
+        MemoryEncryption::Encrypted(51)
+    } else {
+        MemoryEncryption::NoEncryption
+    };
+
     // Safety: these page tables are unused (for now) and we have identity mapping for the lowest 2
```

Comment:
I thought about whether `new` should be marked unsafe or not. I think it shouldn't, because it's not the _creation_ of the `EncryptedPageTable` that's dangerous, it's the _modification_ of active page tables that's dangerous.

That comment also holds for the page table structures in `x86_64`; I don't think `new` should be marked as `unsafe` in them.